src/probe_modules/module_dns.c

```diff
@@ -0,0 +1,1043 @@
+            fs_add_uint64(fs, "dns_rd", dns_hdr->rd); 
+            fs_add_uint64(fs, "dns_tc", dns_hdr->tc); 
+            fs_add_uint64(fs, "dns_aa", dns_hdr->aa); 
+            fs_add_uint64(fs, "dns_opcode", dns_hdr->opcode); 
+            fs_add_uint64(fs, "dns_qr", qr); 
+            fs_add_uint64(fs, "dns_rcode", rcode); 
+            fs_add_uint64(fs, "dns_cd", dns_hdr->cd); 
+            fs_add_uint64(fs, "dns_ad", dns_hdr->ad); 
+            fs_add_uint64(fs, "dns_z", dns_hdr->z); 
+            fs_add_uint64(fs, "dns_ra", dns_hdr->ra); 
+            fs_add_uint64(fs, "dns_qdcount", ntohs(dns_hdr->qdcount)); 
+            fs_add_uint64(fs, "dns_ancount", ntohs(dns_hdr->ancount)); 
+            fs_add_uint64(fs, "dns_nscount", ntohs(dns_hdr->nscount)); 
+            fs_add_uint64(fs, "dns_arcount", ntohs(dns_hdr->arcount)); 
+
+            // And now for the complicated part. Hierarchical data. 
+            char* data = ((char*)dns_hdr) + sizeof(dns_header);
+            uint16_t data_len = udp_len - sizeof(udp_hdr) - sizeof(dns_header);
+            bool err = 0;
+
+            // Questions
+            fieldset_t *list = fs_new_repeated_fieldset();
+            for (int i = 0; i < ntohs(dns_hdr->qdcount) && !err; i++) {
+                err = _process_response_question(&data, &data_len, (char*)dns_hdr, 
+                            udp_len, list);    
+            }
+            fs_add_repeated(fs, "dns_questions", list);
+
+            // Answers
+            list = fs_new_repeated_fieldset();
+            for (int i = 0; i < ntohs(dns_hdr->ancount) && !err; i++) {
+                err = _process_response_answer(&data, &data_len, (char*)dns_hdr, udp_len, list); 
+            }
+            fs_add_repeated(fs, "dns_answers", list);
+
+            // Authorities
+            list = fs_new_repeated_fieldset();
+            for (int i = 0; i < ntohs(dns_hdr->nscount) && !err; i++) {
+                err = _process_response_answer(&data, &data_len, (char*)dns_hdr, udp_len, list);  
+            }
+            fs_add_repeated(fs, "dns_authorities", list);
+
+            // Additionals
+            list = fs_new_repeated_fieldset();
+            for (int i = 0; i < ntohs(dns_hdr->arcount) && !err; i++) {
+                err = _process_response_answer(&data, &data_len, (char*)dns_hdr, udp_len, list);   
+            }
+            fs_add_repeated(fs, "dns_additionals", list);
+
+            // Do we have unconsumed data?
+            fs_add_uint64(fs, "dns_unconsumed_bytes", data_len); 
+
+            if (data_len != 0) {
+                err = 1;
+            }
+
+            // Did we parse OK?
+            fs_add_uint64(fs, "dns_parse_err", err); 
+        }
+    
+        // Now the raw stuff.
+        fs_add_binary(fs, "raw_data", (udp_len - sizeof(struct udphdr)), (void*) &udp_hdr[1], 0);
+        //fs_add_binary(fs, "raw_data", len, (void*)packet, 0);
+   
+        return;
```

Comment:
Yes that is handled. The way it's handled is the top level schema is just a list of things. Then each thing has a schema of an RR. There's no code path that creates differing fields (or if there is it's a bug we didn't hit yet).